### PR TITLE
Use smaller order struct for auction orders

### PR DESCRIPTION
**Pull request by vkgnosis**
In an earlier autopilot PR both Nic and I noticed that it probably doesn't make sense to use the exact same order type for frontend facing orders and auction orders. There is some information that is only relevant for the frontend so the auction order can be smaller. I could also see that in the future the structs diverge more. This PR doesn't do it  but we could also use a separate SQL query for the solvable auction orders that runs more efficiently.

I went through the driver code to see what fields we are using at the moment. This changes OrderMetadata to

```
    pub uid: OrderUid,
    pub owner: H160,
    pub is_liquidity_order: bool,
    pub creation_date: DateTime<Utc>,
    pub executed_amount: U256,
    pub full_fee_amount: U256,
```

from

```
    pub creation_date: DateTime<Utc>,
    pub owner: H160,
    pub uid: OrderUid,
    pub available_balance: Option<U256>,
    pub executed_buy_amount: BigUint,
    pub executed_sell_amount: BigUint,
    pub executed_sell_amount_before_fees: U256,
    pub executed_fee_amount: U256,
    pub invalidated: bool,
    pub status: OrderStatus,
    pub settlement_contract: H160,
    pub full_fee_amount: U256,
    pub is_liquidity_order: bool,
```

.

A lot of the changes in this PR are updating test code that can no longer use OrderBuilder. I felt it wasn't worth it to recreate that machinery for the new orders as struct syntax is good enough.

I'm not entirely sure if the change is worth it but tending towards yes as a form of future proofing and because it makes it more explicit what information is used

### Test Plan

CI

### Release notes

This is a breaking change in the auction api.

### CODE REVIEW

**Comment by MartinquaXD** [comment]
I think overall it makes sense to have specialized structs but I'm worried about the breaking change not being worth the hassle at the moment. Before having such a breaking change we probably have to think about how to pull that off without annoying implementers unnecessarily.

File: crates/model/src/auction.rs [hunk 1]
```diff
@@ -1,11 +1,60 @@
...
+#[serde(rename_all = "camelCase")]
+pub struct OrderMetadata {
+    pub uid: OrderUid,
+    pub owner: H160,
+    pub is_liquidity_order: bool,
+    pub creation_date: DateTime<Utc>,
+    /// sell amount (before fee) for sell order, buy amount for buy order
+    ///
+    /// This field can be U256 while in crate::order::OrderMetadata it is BigRational because the
+    /// other amount could be larger than a U256.
```
Comment (nlordell) [inline crates/model/src/auction.rs:38]: Outdated comment right?

**Review by nlordell — APPROVED** [review]
Big fan of this change.

Agree that we should double check that these fields aren't needed by anyone.

Ideally, we would introduce `api/v2/auction` endpoint for this and still keep the old `v1` with full order data around for a few weeks, but mark it as deprecated in the OpenAPI spec.

**Comment by vkgnosis** [comment]
I just dusted this off because it would be useful for https://github.com/cowprotocol/services/issues/606 . With the new order struct we can add the solver reward into it directly in the auction without also adding it to the main Order struct returned by the api in other routes.

I'm not sure if the extra work of avoiding a breaking change or deprecating is worth it. Deprecation is only useful if people are actually informed about the change through the deprecation and update their code in time. It is unlikely that anyone outside of us is using this api and would actively check the api docs for that.

**Review by sunce86 — APPROVED** [review]

**Comment by vkgnosis** [comment]
Am going to see how much work it is to do it backwards compatible. If it isn't that much will make another PR on top of this one.

**Pull request closed without merging**